### PR TITLE
nu-glob: add fs::symlink_metadata to detect broken symlinks

### PR DESCRIPTION
**Pull request by herlon214**
# Description

Fixes https://github.com/nushell/nushell/issues/3977 (also related to https://github.com/rust-lang-nursery/glob/pull/105)

# Tests

Make sure you've run and fixed any issues with these commands:

- [X] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [X] `cargo clippy --all --all-features -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [X] `cargo build; cargo test --all --all-features` to check that all the tests pass

### CODE REVIEW

**Comment by rgwood** [comment]
Thanks! I'm taking a look now.

Looks like this is generating a warning (can be seen in [the files changed diff](https://github.com/nushell/nushell/pull/5537/files)).

File: crates/nu-glob/src/lib.rs [hunk 1]
```diff
@@ -965,8 +969,9 @@ mod test {
                 .ok()
                 .and_then(|p| match p.components().next().unwrap() {
                     Component::Prefix(prefix_component) => {
-                        let path = Path::new(prefix_component.as_os_str()).join("*");
-                        Some(path)
+                        let path = Path::new(prefix_component.as_os_str());
+                        path.join("*");
```
Comment (rgwood) [inline crates/nu-glob/src/lib.rs:973]: Is this change intentional? From what I can tell line 973 isn't doing anything.

Comment (herlon214) [reply crates/nu-glob/src/lib.rs:973]: Yeah, I just used the same code for https://github.com/rust-lang-nursery/glob/pull/105.
Quite interesting that all CI passed :thinking:

File: crates/nu-glob/src/lib.rs [hunk 2]
```diff
@@ -966,7 +970,7 @@ mod test {
                 .and_then(|p| match p.components().next().unwrap() {
                     Component::Prefix(prefix_component) => {
                         let path = Path::new(prefix_component.as_os_str()).join("*");
-                        Some(path)
+                        Some(path.to_path_buf())
```
Comment (rgwood) [inline crates/nu-glob/src/lib.rs:973]: Nit: why the `to_path_buf()` when it wasn't needed before?

Comment (herlon214) [reply crates/nu-glob/src/lib.rs:973]: Good question, if you check https://github.com/rust-lang-nursery/glob/pull/105/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759L1101 it was there but here looks like @jntrnr removed or at the time it was copied the `to_path_buf()` wasn't present

**Comment by rgwood** [comment]
Thanks!

**Pull request merged**